src/config/remote-config.c

```diff
@@ -316,3 +332,21 @@ STATIC int w_remoted_get_net_protocol(const char * content) {
 
     return retval;
 }
+
+int w_read_tcp_config(XML_NODE node, remoted *logr) {
+    static const char *xml_connection_overtake_time = "connection_overtake_time";
+
+    for (int i = 0; node[i]; i++) {
+        if (!strcmp(node[i]->element, xml_connection_overtake_time)) {
+            if (!OS_StrIsNum(node[i]->content)) {
+                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                minfo("Setting '%s' to default value: '%d'.", node[i]->element, logr->tcp->connection_overtake_time);
+                return OS_SUCCESS;
+            }
+
+            logr->tcp->connection_overtake_time = atoi(node[i]->content);
```

Comment:
I think you should check here is the value is valid (between 0 and 3600) before assigning it. Check for example: https://github.com/wazuh/wazuh/blob/master/src/config/wmodules-agent-upgrade.c#L154

Reply:
Done!